task-list/app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,53 @@
+class TasksController < ApplicationController
+
+
+  def index   #this is enough for the controller to render a view, but need to tell rails when this is an appropriate action
+    @all_tasks = all_incomplete_tasks
+  end
+
+  def all_incomplete_tasks
+    Task.where(completed_at: nil)
+  end
+
+  def show
+    @task = Task.find(params[:id])
+    render :single_task
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    @task = Task.create(task_create_params[:task])
+    redirect_to root_path
+  end
+
+  def destroy
+    Task.find(params[:id]).destroy
+    redirect_to root_path  #redirect does not remember instance variables
+  end
```

Comment:
It may be useful to separate out line 27 into two lines, one line something like @task = Task.find(params[:id]), and the other @task.destroy. That makes it more clear to the reader what's happening. This isn't a big deal for such a small example, but best practice for when you want to chain more than a couple methods together.